pkg/bundle/kube_version.go

```diff
@@ -0,0 +1,37 @@
+package bundle
+
+import (
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/discovery"
+)
+
+type KubeVersionClient interface {
```

Comment:
I wonder if this interface is very useful? It's only got the smallest of changes compared to discovery.ServerVersionInterface, I just don't think it's worth an additional struct, interface, and all that goes with it.

The formatting changes you want for the version can easily be implemented as a pure function: 
```go
formatKubernetesVersion(major, minor string) string {}
```
Adding this interface brings no testability gains, as you can just as easily implement a mock of discovery.ServerVersionInterface as you can one of KubeVersionClient. (I could swear I had written one of those at some point...)

Reply:
Good question, but I'd like to leave this as is for now as this is a bit easier thing for the CLI to use.

Reply:
The CLI isn't using it right now is it? Given that it's new, I suppose it can't be using it at the moment... So why introduce it at all for the CLI?

Reply:
https://github.com/TerryHowe/eks-anywhere-packages/pull/1